cmd/data-scanner.go

```diff
@@ -936,13 +941,17 @@ func (i *scannerItem) applyLifecycle(ctx context.Context, o ObjectLayer, meta ac
 // The resulting size on disk will always be returned.
 // The metadata will be compared to consensus on the object layer before any changes are applied.
 // If no metadata is supplied, -1 is returned if no action is taken.
-func (i *scannerItem) applyActions(ctx context.Context, o ObjectLayer, meta actionMeta) int64 {
+func (i *scannerItem) applyActions(ctx context.Context, o ObjectLayer, meta actionMeta, sizeS *sizeSummary) int64 {
 	applied, size := i.applyLifecycle(ctx, o, meta)
 	// For instance, an applied lifecycle means we remove/transitioned an object
 	// from the current deployment, which means we don't have to call healing
 	// routine even if we are asked to do via heal flag.
-	if !applied && i.heal {
-		size = i.applyHealing(ctx, o, meta)
+	if !applied {
+		if i.heal {
+			size = i.applyHealing(ctx, o, meta)
+		}
+		// replicate only if lifecycle rules are not applied.
+		i.healReplication(ctx, o, meta.oi.Clone(), sizeS)
```

Comment:
this depends on the lifecycle action in question - if it is transition/restore action, then replication still needs to happen

Reply:
why would that happen? @poornas - have we verified such things should happen as per AWS S3?

Reply:
Wait, you're right - I was thinking of situation when object is eligible for transition but somehow sitting in Pending/Failed replication state. Your change will only mean that scanner doesnt do both replication & transition in the same scan - that's ok, except may introduce delay in completing replication